### ISSUE TITLE
Noticable delay before the window showed up

### ISSUE THREAD

**Issue by yvvki** [raphamorim/rio#809]
I have a quite fast NVMe drive and planning to use Rio daily. But unfortunately every time I boot and login to my graphical session, I had a quite annoying delay opening Rio before I can use the terminal.

The other terminal emulator that I have tested include Foot and kitty. While I have no number to compare it against, my personal experience tells me that Foot goes the fastest, Rio is the slowest, while kitty is somewhere in between.

But the actual reason of why I create this issue is that every time I use Rio just after booting, it sometimes have delay of over 2 seconds before showing up the window.

Is there a way to improve Rio start-up time, especially after booting up?

**Comment by raphamorim** [comment]
Hey @yvvki thank you for the issue!

Which backend are you using on Rio? Is the automatic? If yes, I wonder if using Vulkan would speed up

**Comment by yvvki** [comment]
Indeed that switching to Vulkan speed up my system on idle. I use this cute script to help me test the startup time.

```python
import subprocess
import time

def measure_execution_time(command, iterations):
    total_time_ns = 0  # Total time in nanoseconds

    for _ in range(iterations):
        # Record start time
        start_time = time.perf_counter_ns()

        # Execute the command
        subprocess.run(command, shell=True, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL)

        # Record end time
        end_time = time.perf_counter_ns()

        # Calculate elapsed time
        elapsed_time_ns = end_time - start_time
        total_time_ns += elapsed_time_ns

    # Calculate average time in milliseconds
    average_time_ms = total_time_ns / iterations / 1_000_000
    return average_time_ms

# Parameters
iterations = 25
command = "rio -e echo READY"

# Measure and print the average execution time
average_time = measure_execution_time(command, iterations)
print(f"Average execution time over {iterations} runs: {average_time:.2f} ms")
```

I am running on Linux 6.12.0 Zen kernel with 11th Gen Intel i5-1135G7 and this is the result in my machine:

| Backend | Time |
|---|---|
| Automatic | 203.55 ms |
| GL | 202.13 ms |
| Vulkan | 174.25 ms |

I'll try noting down the time just after boot with just 1 iteration ASAP.

**Comment by alkeryn** [comment]
from my testing it's more about the time it takes to open the window somehow, wezterm has the same issue, alacritty doesn't.

it opens an empty window (which copy what was at its place before it was oppened) and after a few seconds, it gets refreshed by the program content.

i found that the issue is not as pronounced when the opened window is smaller.

but if i try to open it on my full 4k screen,  then it takes a while.

**Comment by vikigenius** [comment]
Yep, I think the big problem is, when opening it in full screen there seems to be an initial empty window then it gets populated with the actual terminal.

Interestingly I don't have the issue with WezTerm, but have it with GhosTTY and rio

**Comment by raphamorim** [comment]
these numbers predate the native vulkan backend (0.4.x no longer initializes wgpu on linux by default), worth re-benchmarking on 0.4.11
